src/trie/Trie.ts

```diff
@@ -0,0 +1,169 @@
+import TrieNode from "./TrieNode";
+
+/**
+ * Trie data structure.
+ */
+export default class Trie {
+    private _root: TrieNode;
+    public _lastIndex: number;
+
+    /**
+     * @constructor
+     */
+    constructor() {
+        this._root = new TrieNode(null, true);
+        this._lastIndex = 1;
+    }
+
+    /**
+     * Get root node of the Trie
+     * @returns {TrieNode}
+     */
+    get root(): TrieNode {
+        return this._root;
+    }
+
+    /**
+     * Insert word to the Trie and map data on the word.
+     * If data is not provided it is automatically generated as an increasing number.
+     * @param word - The word to insert.
+     * @param data - Optional data to map to the word.
+     */
+    insert(word: string, data?: any): boolean {
```

Comment:
This class could be defined with a generic type (i.e. `Trie<T>`) so data in a trie can have a type.
The type `any` can be avoided this way.

https://www.typescriptlang.org/docs/handbook/2/generics.html